### PR TITLE
Add parsing and flag for `-bl` for MSbuild-related commands

### PR DESCRIPTION
**Pull request by baronfel**
Fix https://github.com/dotnet/sdk/issues/43310 by explicitly modeling the flag (including argument parsing as described/implemented by MSBuild).

cc @rainersigwald for more MSBuild arg bindings

### CODE REVIEW

File: src/Cli/dotnet/commands/dotnet-run/RunCommand.cs [hunk 1]
```diff
@@ -329,26 +334,21 @@ static ICommand CreateCommandFromRunProperties(ProjectInstance project, RunPrope
                 return command;
             }
 
-            static void InvokeRunArgumentsTarget(ProjectInstance project, string[] restoreArgs, VerbosityOptions? verbosity)
+            static void InvokeRunArgumentsTarget(ProjectInstance project, string[] restoreArgs, BinlogArgs? binlogArgs, VerbosityOptions? verbosity)
```
Comment (baronfel) [inline src/Cli/dotnet/commands/dotnet-run/RunCommand.cs:337]: @mariam-abdulla here's how I used the bl option to correctly forward along binlog information. Add the option to your parser, get the value from the parser, and then use the data structure to create the `BinaryLogger` you need (if any).

File: src/Cli/dotnet/CommonOptions.cs [hunk 2]
```diff
@@ -180,6 +198,57 @@ internal static string ArchOptionValue(ParseResult parseResult) =>
 
         public static readonly CliOption<string[]> TestLoggerOption = new("--logger");
 
+        public static readonly CliOption<BinlogArgs?> BinaryLoggerOption = new ForwardedOption<BinlogArgs?>("-bl", "--bl")
+        {
+            Arity = ArgumentArity.ZeroOrOne,
+            //TODO: loc
```
Comment (nagilson) [inline src/Cli/dotnet/CommonOptions.cs:204]: This loc step should be done here.

Comment (baronfel) [reply src/Cli/dotnet/CommonOptions.cs]: Ok I added loc just for you

**Comment by baronfel** [comment]
This is what I was afraid of - we can't define `-bl` in a safe way it seems with System.CommandLine.

-bl can be supplied as a flag and as an option that has some configuration: a binlog filename to create and/or a directive how to manage the files that are imported into the binlog. We can't enforce the MSbuild syntax requirement here of `-bl:<stuff>` because System.CommandLine also allows `-bl <stuff>`. We can't use the `OnlyTake` mechanism inside the option for some implementation reason. So we're stuck here and what we'll have to do for `run` and `test` is fall back to manual scraping of the `string[]` of unbound args.

**Pull request closed without merging**